system/security/shared/caching.hpp

```diff
@@ -163,6 +164,29 @@ class CPermissionsCache
         return instance;
     }
 
+    //Returns a shared cache of a given type
+    //Call this method with a unique class string ("LDAP", "MyOtherSecMgr")
+    //to create a cache shared amongst security managers of the same class
+    static CPermissionsCache* queryInstance(const char * secMgrClass)
+    {
+        if (secMgrClass == nullptr)
+            return queryInstance();
+
+        typedef map<string, CPermissionsCache*> MapCache;
```

Comment:
knee jerk reaction was to ask to move typedef declaration outside of the method, doesn't hurt to make it visible to others. But is there a reason to keep it encapsulated? (other than nobody else is using it right now)